sys/include/isrpipe.h

```diff
@@ -54,10 +54,22 @@ typedef struct {
  */
 void isrpipe_init(isrpipe_t *isrpipe, uint8_t *buf, size_t bufsize);
 
+/**
+ * @brief   Put number of characters into the isrpipe's buffer
+ *
+ * @param[in]   isrpipe     isrpipe object to operate on
+ * @param[in]   buf         characters to add to isrpipe buffer
+ * @param[in]   n           number of characters to add from buf to isrpipe's buffer
+ *
+ * @returns     number of characters that could be added
+ * @returns     -1 if buffer was full
+ */
+int isrpipe_write(isrpipe_t *isrpipe, uint8_t *buf, size_t n);
```

Comment:
I think we should split this one and add a unittest.

Reply:
I opened #17336 for this change. I think the unit test for `tsrb` could be already enough.